### PR TITLE
[0.6.x] Backport Macos timer from #2342

### PR DESCRIPTION
**Pull request by Mrcubix**
Backport timer changes from #2342 using the worst git usage skills known to man.

### CODE REVIEW

**Comment by Mrcubix** [comment]
I hate the linter.

**Comment by Mrcubix** [comment]
I'm getting lint error from stuff done in the original PR, send help.

**Comment by Mrcubix** [comment]
It's still complaining about import ordering, i'm gonna loose my mind

**Comment by Mrcubix** [comment]
Since the linter doesn't specify which import is causing an issue, i'm unable to fix it 
(especially since i reversed the static & defined import and the issue still occurs)

File: OpenTabletDriver.Desktop/Interop/Timer/MacOSTimer.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using OpenTabletDriver.Native.Linux;
+using OpenTabletDriver.Native.OSX.Timers;
+using OpenTabletDriver.Plugin;
+using OpenTabletDriver.Plugin.Timers;
+using Thread = System.Threading.Thread;
+
+using static OpenTabletDriver.Native.Posix.Utility;
+using static OpenTabletDriver.Native.OSX.Timers.Timers;
```
Comment (Kuuuube) [inline OpenTabletDriver.Desktop/Interop/Timer/MacOSTimer.cs:11]:
```suggestion
using System;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;
using OpenTabletDriver.Native.Linux;
using OpenTabletDriver.Native.OSX.Timers;
using OpenTabletDriver.Plugin;
using OpenTabletDriver.Plugin.Timers;
using static OpenTabletDriver.Native.OSX.Timers.Timers;
using static OpenTabletDriver.Native.Posix.Utility;
using Thread = System.Threading.Thread;
```

Comment (Mrcubix) [reply OpenTabletDriver.Desktop/Interop/Timer/MacOSTimer.cs:11]: it complained about that exact format in 406225bf7c8eac91c00e61fb048fac755a2e194a

Comment (Kuuuube) [reply OpenTabletDriver.Desktop/Interop/Timer/MacOSTimer.cs]: Not the same `using System.Linq;` was there when it wasnt supposed to be in that commit. You can run `dotnet format OpenTabletDriver.sln` yourself if you want. It's what the linter is checking against to see if formatting is bad.

Comment (Mrcubix) [reply OpenTabletDriver.Desktop/Interop/Timer/MacOSTimer.cs]: Sure, it's the only difference, and that should throw an ordering error

Comment (Kuuuube) [reply OpenTabletDriver.Desktop/Interop/Timer/MacOSTimer.cs]: Microsoft being funny with the output of `--verify-no-changes`.

Comment (Mrcubix) [reply OpenTabletDriver.Desktop/Interop/Timer/MacOSTimer.cs]: image

Comment (Mrcubix) [reply OpenTabletDriver.Desktop/Interop/Timer/MacOSTimer.cs]: I see it now, but still that linter doesn't cut it to know which one is the source of the issue & i shouldn't have to run a command that for some reasons runs for a whole 2 minutes just for a linting error that wouldn't be thrown by any other sane linter.

**Comment by Mrcubix** [comment]
Forgot to specify : 
Closes #3389

but of course someone else is gonna need to add it manually and merging won't close it (happened to me a week ago)

**Comment by jamesbt365** [comment]
Theres no point linking it, we will need to manually close it anyway because 0.6.x is not the default branch of the repository.

**Review by gonX — COMMENTED** [review]
Code changes match original PR (with appropriate changes for `0.6.x` codebase), but would like some testing from a MacOS user.

/e: in less convoluted words: LGTM, would like testing though

**Comment by AkiSakurai** [comment]
Tested on intel mac, filter is working

**Review by gonX — APPROVED** [review]
Approving given AkiSakurai's testing

**Review by jamesbt365 — APPROVED** [review]

**Pull request merged**